packages/ruleset/test/path-segment-case-convention.test.js

```diff
@@ -0,0 +1,63 @@
+const { pathSegmentCaseConvention } = require('../src/rules');
+const { makeCopy, rootDocument, testRule, severityCodes } = require('./utils');
+
+const rule = pathSegmentCaseConvention;
+const ruleId = 'path-segment-case-convention';
+const expectedSeverity = severityCodes.error;
+
+describe('Spectral rule: path-segment-case-convention', () => {
+  describe('Should not yield errors', () => {
+    it('Clean spec', async () => {
+      const results = await testRule(ruleId, rule, rootDocument);
+      expect(results).toHaveLength(0);
+    });
+
+    it('Empty path segment', async () => {
+      const testDocument = makeCopy(rootDocument);
+
+      testDocument.paths['/v1//drinks//last_seg'] =
+        testDocument.paths['/v1/drinks'];
+      delete testDocument.paths['/v1/drinks'];
+
+      const results = await testRule(ruleId, rule, testDocument);
+      expect(results).toHaveLength(0);
+    });
+  });
+
+  describe('Should yield errors', () => {
+    it('Path segments are camel case', async () => {
+      const testDocument = makeCopy(rootDocument);
+
+      testDocument.paths['/v1/thisIsATest/drinks/anotherTest'] =
+        testDocument.paths['/v1/drinks'];
+      delete testDocument.paths['/v1/drinks'];
+
+      const results = await testRule(ruleId, rule, testDocument);
+      expect(results).toHaveLength(2);
+      for (const result of results) {
+        expect(result.code).toBe(ruleId);
+        expect(result.message).toMatch(/^Path segments must be snake case:/);
+        expect(result.severity).toBe(expectedSeverity);
+        expect(result.path.join('.')).toBe(
+          'paths./v1/thisIsATest/drinks/anotherTest'
+        );
+      }
+    });
+    it('Path segment has a .', async () => {
+      const testDocument = makeCopy(rootDocument);
+
+      testDocument.paths['/v1/bad.segment/drinks'] =
+        testDocument.paths['/v1/drinks'];
+      delete testDocument.paths['/v1/drinks'];
+
+      const results = await testRule(ruleId, rule, testDocument);
+      expect(results).toHaveLength(1);
+      for (const result of results) {
+        expect(result.code).toBe(ruleId);
+        expect(result.message).toMatch(/^Path segments must be snake case:/);
```

Comment:
I think it would be good for at least this test to verify the full message you define in the code (with the name of the segment, etc.) since it isn't necessary for it to be contained in the loop

Reply:
fixed it